### PR TITLE
Feature/share modal

### PR DESCRIPTION
**Pull request by clementprdhomme**
This PR create the basic modal component and opens one when the map's share button is clicked. Additionally, it changes one rule of sass-lint to enable the use of the `rgba` function outside of variables and adds a basic `_settings.scss` file.

### CODE REVIEW

File: app/src/components/shared/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    this.onKeyPress = e => {
+      if (e.keyCode !== 27) return;
+      e.preventDefault();
+      this.props.close();
+    };
+
+    /* If the modal is opened at instantiation, we want the keypress handler to be active */
+    if (props.opened) {
+      document.addEventListener('keypress', this.onKeyPress);
```
Comment (nerik) [inline app/src/components/shared/Modal.jsx]: This doesn't work in Chrome (51). Use `'keydown'` instead

File: app/src/components/shared/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  render() {
+    if (!this.props.opened) return null;
+
+    return (
+      <div className={styles['c-modal']} onClick={(e) => this.onClickOverlay(e)}>
+        <div className={styles.content}>
+          <button className={styles['close-button']} onClick={() => this.props.close()}>
+            <svg className={styles.icon} title="Close this modal">
+              <g strokeWidth="3" stroke="#43649C" fill="none" fillRule="evenodd" strokeLinecap="square">
+                <path d="M2.627 2.627L13.43 13.43M13.373 2.627L2.57 13.43" />
```
Comment (nerik) [inline app/src/components/shared/Modal.jsx]: We need to find a solution for this. Might be doable without webpack?

Comment (nerik) [inline app/src/components/shared/Modal.jsx]: https://github.com/matthewwithanm/react-inlinesvg ?

File: app/src/components/shared/Modal.jsx (hunk 1 above)
Comment (clementprdhomme) [inline app/src/components/shared/Modal.jsx]: Good catch, reference here: https://www.w3.org/TR/uievents/#event-type-keypress

> If supported by a user agent, this event must be dispatched when a key is pressed down, **if and only if that key normally produces a character value**.

File: app/src/components/shared/Modal.jsx [hunk 3]
```diff
@@ -42,11 +44,7 @@ class Modal extends React.Component {
       <div className={styles['c-modal']} onClick={(e) => this.onClickOverlay(e)}>
         <div className={styles.content}>
           <button className={styles['close-button']} onClick={() => this.props.close()}>
-            <svg className={styles.icon} title="Close this modal">
-              <g strokeWidth="3" stroke="#43649C" fill="none" fillRule="evenodd" strokeLinecap="square">
-                <path d="M2.627 2.627L13.43 13.43M13.373 2.627L2.57 13.43" />
-              </g>
-            </svg>
+            <Icon className={styles.icon} title="Close this modal" />
```
Comment (nerik) [inline app/src/components/shared/Modal.jsx]: Awesome

**Pull request merged**